packages/good-design/src/apps/bridge/MicroBridge.tsx

```diff
@@ -134,48 +72,40 @@ export const MicroBridge = ({
   onBridgeStart,
   onBridgeFailed,
   onBridgeSuccess
-}: {
-  onBridge: OnBridge;
-  useBalanceHook: (chain: "fuse" | "celo") => string;
-  useCanBridge: (chain: "fuse" | "celo", amountWei: string) => { isValid: boolean; reason: string };
-  onSetChain?: (chain: "fuse" | "celo") => void;
-  limits?: ILimits;
-  fees?: IFees;
-  bridgeStatus?: Partial<TransactionStatus>;
-  relayStatus?: Partial<TransactionStatus>;
-  selfRelayStatus?: Partial<TransactionStatus>;
-  onBridgeStart?: () => void;
-  onBridgeSuccess?: () => void;
-  onBridgeFailed?: (e: Error) => void;
-}) => {
+}: MicroBridgeProps) => {
   const [isBridging, setBridging] = useState(false);
-  const [inputWei, setInput] = useState<string>("0");
-
-  const [sourceChain, setSourceChain] = useState<"fuse" | "celo">("fuse");
-
+  const { nextStep } = useWizard();
+  const [sourceChain, setSourceChain] = originChain;
   const targetChain = sourceChain === "fuse" ? "celo" : "fuse";
-  const balanceWei = useBalanceHook(sourceChain);
-
-  const { isValid, reason } = useCanBridge(sourceChain, inputWei);
-  const hasBalance = Number(inputWei) <= Number(balanceWei);
+  const [toggleState, setToggleState] = useState<boolean>(false);
+
+  const balances = useBalanceHook() ?? {};
+  const { wei, gdValue } = balances[sourceChain] ?? {};
+  const [bridgeWeiAmount, setBridgeAmount] = inputTransaction;
+  const [, setPendingTransaction] = pendingTransaction;
+  const { isValid, reason } = useCanBridge(sourceChain, bridgeWeiAmount);
+  const { minimumAmount, expectedToReceive } = useBridgeEstimate({
+    limits,
+    fees,
+    inputWei: bridgeWeiAmount,
+    sourceChain
+  });
+
+  const hasBalance = Number(bridgeWeiAmount) <= Number(wei);
   const isValidInput = isValid && hasBalance;
   const reasonOf = reason || (!hasBalance && "balance") || "";
 
   const toggleChains = useCallback(() => {
     setSourceChain(targetChain);
+    setToggleState(prevState => !prevState);
     onSetChain?.(targetChain);
   }, [setSourceChain, onSetChain, targetChain]);
 
   const triggerBridge = useCallback(async () => {
-    setBridging(true);
+    setPendingTransaction({ bridgeWeiAmount, expectedToReceive });
     onBridgeStart?.();
-
-    try {
-      await onBridge(inputWei, sourceChain);
-    } finally {
-      setBridging(false);
-    }
-  }, [setBridging, onBridgeStart, onBridge, inputWei, sourceChain]);
+    void nextStep();
```

Comment:
shouldnt this be a condition that the onBridgeStart actually did some kind of TX?
what happens if the user rejects the bridge TX on metamask?

Reply:
And the condition? No, we track the state of the transaction on the next step (with the sign / confirmation modals)

Reply:
so what happens if the user actually reject the TX?
it will go to the nextStep()
what will happen then and how? the onError callback?

Reply:
https://github.com/GoodDollar/GoodWeb3-Mono/blob/42ab4b093318660dc45469dba788a902b00b0acd/packages/good-design/src/apps/bridge/MicroBridgeController.tsx#L316